common/app/targeting/CampaignAgent.scala

```diff
@@ -0,0 +1,27 @@
+package targeting
+
+import common._
+import com.gu.targeting.client.CampaignCache
+import conf.Configuration
+import scala.util.control.NonFatal
+import scala.concurrent.Future
+
+object CampaignAgent extends Logging with ExecutionContexts {
+  private val agent = AkkaAgent[CampaignCache](CampaignCache(List()))
+
+  def refresh(): Future[Unit] = {
+    Configuration.targeting.campaignsUrl.map(url => {
+      CampaignCache.fetch(url).map(agent.send)
+    }).getOrElse(Future.failed(throw new BadConfigurationException("Campaigns URL not configured")))
```

Comment:
I think agent.send is returning a Unit here?  you might want to use something that returns a future, and then do a Future.sequence on the result.  Otherwise the scheduler will know when the CampaignCache.fetch completes, but won't realise that the agent.send is still in progress.